pydgraph/client_slash.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 Dgraph Labs, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Slash GraphQL python client."""
+import grpc
+
+import pydgraph
+from pydgraph import txn, util, DgraphClientStub
+from pydgraph.meta import VERSION
+from pydgraph.proto import api_pb2 as api
+
+__author__ = 'Anurag Sharma <anurag@dgraph.io>'
+__maintainer__ = 'Arpan Gupta <arpan@dgraph.io>'
+__version__ = VERSION
+__status__ = 'development'
+
+
+class SlashGraphQLClient(object):
+    """Creates a new Client for interacting with the Slash GraphQL.
+    """
+
+    def __init__(self, slash_end_point, api_key):
+        if not slash_end_point:
+            raise ValueError('No Slash endpoint provided in SlashGraphQLClient constructor')
+
+        self._slash_end_point = slash_end_point
+        self._api_key = api_key
+        self._port = "443"
+
+    def get_stub(self):
+        """Returns Dgraph Client stub for the Slash GraphQL endpoint"""
+        url = self._slash_end_point.replace("/graphql", "")
```

Comment:
I think you should accept multiple formats here:
1) The Slash GraphQL endpoint: "https://foo.aws.cloud.dgraph.io/graphql"
2) Any url on that domain: "https://foo.as.cloud.dgraph.io"
3) A String in the format "foo.grpc.aws.cloud.dgraph.io:443"

Reply:
Added the support for 1. and 2. Not sure if we want to support 3. as well, that's not a valid URL as per URL-parser. Would have to write few lines separately to handle that. Let's keep it simple and support the graphql endpoint which we get from dashboard only?

Reply:
ok, looks good